app/shared/src/commonMain/kotlin/ui/subject/episode/video/settings/DanmakuRegexFilterGroup.kt

```diff
@@ -0,0 +1,372 @@
+import androidx.compose.ui.window.Dialog
+import me.him188.ani.app.ui.external.placeholder.placeholder
+import me.him188.ani.app.ui.foundation.ifThen
+import me.him188.ani.app.ui.foundation.text.ProvideTextStyleContentColor
+import me.him188.ani.app.ui.settings.framework.components.SettingsScope
+import me.him188.ani.danmaku.ui.DanmakuRegexFilter
+import me.him188.ani.danmaku.ui.DanmakuRegexFilterConfig
+import java.util.regex.Pattern
+import java.util.regex.PatternSyntaxException
+
+internal fun isValidRegex(pattern: String): Boolean {
+    return try {
+        Pattern.compile(pattern)
+        true
+    } catch (e: PatternSyntaxException) {
+        false
+    }
+}
+
+@Composable
+internal fun SettingsScope.DanmakuRegexFilterGroup(
+    danmakuRegexFilterConfig: DanmakuRegexFilterConfig,
+    addDanmakuRegexFilter: (filter: DanmakuRegexFilter) -> Unit,
+    editDanmakuRegexFilter: (filter: DanmakuRegexFilter) -> Unit,
+    removeDanmakuRegexFilter: (filter: DanmakuRegexFilter) -> Unit,
+    switchDanmakuRegexFilter: (fiter: DanmakuRegexFilter) -> Unit,
+    isLoadingState: Boolean
+) {
+    var showAdd by remember { mutableStateOf(false) }
+    var showError by remember { mutableStateOf(false) }
+    var errorMessage by remember { mutableStateOf("") }
+
+    if (showAdd) {
+        AddRegexFilterDialog(
+            onDismissRequest = { showAdd = false },
+            onConfirm = { name, regex ->
+                if (name.isBlank() || regex.isBlank()) {
+                    errorMessage = "名字和正则不能为空"
+                    showError = true
+                } else {
+                    if (isValidRegex(regex).not()){
+                    errorMessage = "正则输入法不正确"
+                    showError = true 
+                    } else { 
+                        addDanmakuRegexFilter(DanmakuRegexFilter(name = name, re = regex))
+                        showAdd = false
+                    }
+                }
+            },
+            title = { Text("添加正则过滤器") },
+            description = {
+                Text("请正确添加正则表达式，例：第一个字符为数字：'^[1-9]{1}\$'.\n符合正则的弹幕讲不会被显示")
+            },
+            
+        )
+    }
+
+    if (showError) {
+        AlertDialog(
+            onDismissRequest = { showError = false },
+            confirmButton = {
+                TextButton(onClick = { showError = false }) {
+                    Text("确认")
+                }
+            },
+            title = { Text("名字和正则不能为空") },
+            text = { Text(errorMessage) }
+        )
+    }
+    
+    Group(
+        title = { Text("正则过滤器管理") },
+        actions = {
+            Row {
+                IconButton({
+                    showAdd = true
+                }) {
+                    Icon(Icons.Rounded.Add, contentDescription = "添加正则")
+                }
+            }
+        }
+    ) {
+        FlowRow (
+            Modifier.placeholder(isLoadingState).fillMaxWidth(),
+            horizontalArrangement = Arrangement.spacedBy(14.dp)
```

Comment:
```suggestion
            horizontalArrangement = Arrangement.spacedBy(14.dp)
```
用标准的 8, 12, 16 dp 吧?

Reply:
@StageGuard 有说法么

Reply:
https://m3.material.io/foundations/layout/understanding-layout/spacing
m3 guideline 确实提到用 4 的倍数 padding，最好是用吧